packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -33,17 +35,23 @@ class SharedPreferences {
   /// previous behavior of this plugin. To use preferences with no prefix,
   /// set [prefix] to ''.
   ///
+  /// If [prefix] is set to '', you may encounter preferences that are
+  /// incompatible with shared_preferences. The optional parameter
+  /// [allowList] will force the plugin to only return preferences that
+  /// are contained in the list AND match the provided prefix.
+  ///
   /// No migration of existing preferences is performed by this method.
   /// If you set a different prefix, and have previously stored preferences,
   /// you will need to handle any migration yourself.
   ///
   /// This cannot be called after `getInstance`.
-  static void setPrefix(String prefix) {
+  static void setPrefix(String prefix, {Set<String>? allowList}) {
```

Comment:
I am guessing it is overkill but did you consider making this a function that returned a bool instead of a set?

Reply:
We need to do the filtering on the native side, so having it be a Dart function would make thing *way* more complicated.

Reply:
ok that makes sense.